src/java.base/share/classes/java/util/Timer.java

```diff
@@ -588,7 +588,9 @@ private void mainLoop() {
                             // A zero checkpointRestoreTimeDelta value indicates no Checkpoint performed yet,
                             // it can't be negative, otherwise a RestoreException already was thrown.
                             if (checkpointRestoreTimeDelta > 0) {
+                                // Ensure the time compensation in milliseconds is not less than the checkpoint restore time delta.
                                 task.nextExecutionTime += (checkpointRestoreTimeDelta / 1000000);
+                                task.nextExecutionTime += ((checkpointRestoreTimeDelta % 1000000 == 0) ? 0 : 1);
```

Comment:
actually you could just do `task.nextExecutionTime += (((checkpointRestoreTimeDelta % 1000000 == 0) & 1`

Reply:
the above doesnt work